src/amo/components/AutoSearchInput/index.js

```diff
@@ -151,6 +152,13 @@ export class AutoSearchInputBase extends React.Component<InternalProps, State> {
       delete filtersFromLocation.page;
     }
 
+    if (
+      filtersFromLocation.sort &&
+      filtersFromLocation.sort === SEARCH_SORT_RANDOM
+    ) {
+      delete filtersFromLocation.sort;
```

Comment:
It's not obvious why this is getting deleted so I suggest putting a comment here.